tests/manual/controls/SelectTableViewTest.java

```diff
@@ -0,0 +1,137 @@
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TableView;
+import javafx.scene.layout.BorderPane;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+public class SelectTableViewTest extends Application {
+
+    final int ROW_COUNT = 700_000;
+    //  final int ROW_COUNT = 80_000;
+    //  final int ROW_COUNT = 50_000;
+    //  final int ROW_COUNT = 8_000;
+    final int COL_COUNT = 3;
+
+    @Override
+    public void start(Stage stage) {
+        TableView<String[]> tableView = new TableView<>();
+        tableView.getSelectionModel().setSelectionMode(SelectionMode.MULTIPLE);
+//      tableView.getSelectionModel().setSelectionMode(SelectionMode.SINGLE);
+
+        final ObservableList<TableColumn<String[], ?>> columns = tableView.getColumns();
+        for(int i = 0; i < COL_COUNT; i++) {
+            TableColumn<String[], String> column = new TableColumn<>("Col"+i);
+            final int colIndex=i;
+            column.setCellValueFactory(cell -> new SimpleStringProperty(cell.getValue()[colIndex]));
+            column.setPrefWidth(150);
+            columns.add(column);
+        }
+
+        ObservableList<String[]> items = tableView.getItems();
+        for(int i = 0; i < ROW_COUNT; i++) {
+            String[] rec = new String[COL_COUNT];
+            for(int j = 0; j < rec.length; j++) {
+                rec[j] = i + ":" + j;
+            }
+            items.add(rec);
+        }
+
+        BorderPane root = new BorderPane(tableView);
+        Button selectAll = new Button("selectAll");
+        Button clearSelection = new Button("clearSelection");
+        Button selectToStart = new Button("selectToStart");
+        Button selectToEnd = new Button("selectToEnd");
+        Button selectPrevious = new Button("selectPrevious");
+        Button selectNext= new Button("selectNext");
+
+        selectAll.setFocusTraversable(true);
+        clearSelection.setFocusTraversable(true);
+        selectToStart.setFocusTraversable(true);
+        selectToEnd.setFocusTraversable(true);
+        selectPrevious.setFocusTraversable(true);
+        selectNext.setFocusTraversable(true);
+
+        root.setRight(new VBox(6, selectAll, selectToStart, selectToEnd, selectPrevious, selectNext, clearSelection));
+        stage.setScene(new Scene(root, 600, 600));
+
+        selectAll.setOnAction(e -> selectAll(tableView));
+        clearSelection.setOnAction(e -> clearSelection(tableView));
+        selectToStart.setOnAction(e -> selectToStart(tableView));
+        selectToEnd.setOnAction(e -> selectToLast(tableView));
+        selectPrevious.setOnAction(e -> selectPrevious(tableView));
+        selectNext.setOnAction(e -> selectNext(tableView));
+
+        stage.show();
+    }
+
+    private void selectAll(TableView tableView) {
+        long t = System.currentTimeMillis();
+        tableView.getSelectionModel().selectAll();
+        System.out.println("time:"+ (System.currentTimeMillis() - t));
```

Comment:
Minor: space before the `+` here and a few places below (also in the other test as noted).